### DISCUSSION TITLE
Documentation on nanomq-sqlite for Linux not working

### DISCUSSION THREAD

**Discussion by parkershute** [nanomq/nanomq#1442] [General]
I have been following the documentation for installing the nanomq-sqlite version but it is not working.

It states very clearly here: https://nanomq.io/docs/en/latest/installation/packages.html#install-nanomq

"**You can select the appropriate installation version of NanoMQ based on your specific business requirements. To do this, simply replace nanomq in the installation command with the corresponding version code:

For the SQLite Edition of NanoMQ, use nanomq-sqlite**"

I am using Debian and following the APT instructions using the curl for the repo.

The instructions seem to be clear on this process but it is not working. 

What am I missing?

![image](https://github.com/emqx/nanomq/assets/30698285/3d5fd4b8-e3ed-47ff-8033-68b7801cd368)

**Comment by JaylinYu** [comment]
``nanomq-sqlite`` version refers to here https://github.com/emqx/nanomq/releases/tag/0.19.5

I dont think they pushed all released file to apt source.

**Reply by parkershute** [reply]
Thank you JaylinYu. Is there a plan to push the releases for apt? I need something that will be scalable across a lot of devices.

**Comment by rommni** [comment]
I have the same problem with nanomq-full, is there a plan to made all versions accessible with apt or should we change the documentation ?

**Reply by JaylinYu** [reply]
Recommend to download the full ver of binary package.
Or compile And install from source

**Comment by calliday** [comment]
This is the documentation that needs to be updated: https://nanomq.io/docs/en/latest/installation/packages.html#linux

It claims that all you have to do from a Debian machine is use apt to install `nanomq-sqlite`, `nanomq-full` or whichever version you want, but if I'm understanding correctly, we have to build from source to get SQLite functionality. @JaylinYu is that correct?

**Reply by JaylinYu** [reply]
Hi bro, Both way works, either install `nanomq-sqlite/nanomq-full` version, or build from source with `-DNNG_ENABLE_SQLITE=ON`.
DDS, SOME/IP and AWS bridging is where you have to start from building from source, it is an enterprise feature

**Reply by calliday** [reply]
Thanks for your quick response! Unfortunately, on my Debian Bullseye machine, requesting either `nanomq-sqlite` or `nanomq-full` is resulting in `E: Unable to locate package nanomq-sqlite`. I can retrieve `nanomq` and get version `0.21.8` but I can't find either the SQLite or full version, even for older releases. Any ideas what could be happening here?

**Reply by JaylinYu** [reply]
Unfortunately, NanoMQ is not available on official APT source... That requires a long licensing process. You have to download the binary package from github resource and install Via "dpkg -i"

**Reply by calliday** [reply]
Thank you for your help, the long license process is understandable.

However, using the custom repo defined in NanoMQ's documentation, I _can_ get `nanomq` using APT. I'm using the [custom apt source](https://nanomq.io/docs/en/latest/installation/packages.html#install-with-apt-source) listed in the [Linux NanoMQ Installation](https://nanomq.io/docs/en/latest/installation/packages.html#linux) documentation. Using this, I'm able add the custom repository and `apt install nanomq`. However, to get `nanomq-sqlite`, I must grab the debian package myself, like you mention.

It makes sense that the APT maintenance would be more trouble than it's worth. If we could get the docs updated to reflect this package availability, it would help save developers time trying to figure out how to get (and automate getting) the version of the packages they need.
